File: Data Structures & Algorithms/merge-strings-alternately/submission-0.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        min_len = min(len(word1), len(word2))
        max_len = max(len(word1), len(word2))
        res = ""
        i = 0
        while i < min_len:
            res += word1[i]
            res += word2[i]
            i += 1

        while i < len(word1):
            res += word1[i]
            i += 1
        while i < len(word2):
            res += word2[i]
            i += 1
            
        return res

File: Data Structures & Algorithms/longest-turbulent-subarray/submission-0.cpp
class Solution {
public:
    int maxTurbulenceSize(vector<int>& arr) {
        int n = arr.size();
        int count = 0;
        int res = 0;
        int sign = -1;

        for (int i = 0; i < arr.size()-1; i++) {
            if (arr[i] > arr[i+1]) {
                if (sign == 0) {
                    count++;
                } else {
                    count = 1;
                }
                sign = 1;
            } else if (arr[i] < arr[i+1]) {
                if (sign == 1) {
                    count++;
                } else {
                    count = 1;
                }
                sign = 0;
            } else {
                count = 0;
                sign = -1;
            }

            res = max(res,count);
        }

        return res + 1;
    }
};